### ISSUE TITLE
difference between "who" and "whom"

### ISSUE THREAD

**Issue by s3hMC** [languagetool-org/languagetool#3034]
The title is pretty self-explanatory. LanguageTool should tell the difference between the words “who” and “whom.”

**Comment by tiff** [comment]
We already have a few rules for this. Could you provide some example sentences, where who/whom were incorrectly used? Thanks

**Comment by s3hMC** [comment]
The only incorrect sentence type that is not detected by LanguageTool that I could find is this one:
`John Kennedy, whom is dead.`

**Comment by bebras123** [comment]
Possible correction:
<details>
<summary>Updated WHOM_WHO rule</summary>

```xml
<rule id="WHOM_WHO" name="whom (who)">                                            
    <antipattern>
        <token skip='-1'>whom</token>
        <token postag="IN"/>
        <token postag="SENT_END"/>
    </antipattern>
    <antipattern>
        <token postag="IN"/>
        <token regexp='yes'>whom(?:ever)?</token>
    </antipattern>
    <antipattern>
        <token regexp='yes'>whom(?:ever)?</token>
        <token min="0" inflected="yes" regexp="yes">do|be</token>
        <token postag="PRP|DT|NNP" postag_regexp="yes"></token>
    </antipattern>
    <antipattern>
        <token regexp='yes'>whom(?:ever)?</token>
        <token min="0" inflected="yes" regexp="yes">do|be</token>
        <token postag="NN.*" postag_regexp="yes"><exception postag="NN.*" postag_regexp="yes" negate_pos="yes"></exception></token>
    </antipattern>                                            
    <antipattern>
        <token regexp='yes'>whom(?:ever)?</token>
        <token inflected="yes" regexp="yes">do|be</token>
        <token>one</token>
    </antipattern>
    <pattern>
        <marker>
            <token regexp='yes'>whom(?:ever)?</token>
        </marker>
        <token postag="VB[ZD]" postag_regexp="yes"/>
    </pattern>
    <message>Did you mean <suggestion><match no="1" regexp_match="hom" regexp_replace="ho"/></suggestion> instead of 'whom'?</message>
    <url>https://www.dictionary.com/e/whomever-vs-whoever/</url>
    <example correction="Who"><marker>Whom</marker> paid for the meal?</example>
    <example correction="who">He is demanding £5,000 from the elderly woman <marker>whom</marker> has ruined his life.</example>
    <example>With whom did you talk?</example>
    <example>Whom did you meet?</example>
    <example>The hall was filled with students, many of whom were girls.</example>
    <example>Whom are you talking to?</example><!-- Common colloquial speech uses 'Who are you talking to?" incorrectly: https://forum.wordreference.com/threads/who-whom-you-are-talking-to.512760/ 2020-02-10 MFU Lexico says that 'who' is now generally accepted: www.lexico.com/definition/who -->
    <example>Legion of Whom?</example>
    <example correction="who">John Kennedy, <marker>whom</marker> is dead.</example>
    <example correction="who">If this is still confusing, Please have someone <marker>whom</marker> is fluent (a personl able to express oneself easily) in American English review it.</example>
    <example correction="whoever">It is not a guarantee that <marker>whomever</marker> answers you will know enough to know when the names you’re using are Mac-specific.</example>
    <example correction="who">It is really hard to notice <marker>whom</marker> is talking to you.</example>
    <example correction="Whoever"><marker>Whomever</marker> pays the price</example>
    <example>Whom is the speech trying to convince?</example>
    <example>Moving on, whom does one ask to rate the sense of humour of an individual?</example>
    <example>Ayaz said her that whom is she waiting?</example>
    <example correction="who">These women had little to no difference in infant birth weight compared to the women <marker>whom</marker> did not undergo amalgam restoration during pregnancy.</example>
    <example>Whom did Larry send to the store?</example>
    <example>Whom is he trying to avoid?</example>
    <example>Todd, whom Brooks called the "evil 'other, was the leader and most villainous of the group and had a previous physical relationship with Marty.</example>
</rule>

```
</details>

**Comment by MikeUnwalla** [comment]
@bebras123 , thank you.

I kept the rulegroup, because possibly, we will need other related rules.

I made some formatting changes for consistency with our current coding style:
- Double quotes.
- Exceptions on separate lines.
- Where possible, compact format for end tags for elements (example: `/>`, not `</token>`).

We don't have a good way to track changes. I think it is useful to link an item of work with an issue by adding a comment that shows the issue number.

I changed the order of the examples such that all the incorrect examples come first, then all the correct examples. Just personal preference.
